src/View/Widgets/BreadcrumbsEntry.vala

```diff
@@ -319,7 +319,7 @@ namespace Marlin.View.Chrome {
             if (!drop_data_ready) {
                 drop_file_list = null;
                 foreach (var uri in selection_data.get_uris ()) {
-                    drop_file_list.append (File.new_for_uri (uri));
+                    drop_file_list.prepend (File.new_for_uri (uri));
                     drop_data_ready = true;
```

Comment:
Dropping onto a breadcrumb adds files to a directory which is not sensitive to the order, so we can prepend without reversing.